src/frontends/tensorflow/src/op/reverse.cpp

```diff
@@ -3,47 +3,126 @@
 //
 
 #include "op_table.hpp"
-#include "openvino/opsets/opset8.hpp"
+#include "openvino/opsets/opset9.hpp"
 
 using namespace std;
-using namespace ov::opset8;
+using namespace ov;
+using namespace ov::opset9;
 
 namespace ov {
 namespace frontend {
 namespace tensorflow {
 namespace op {
+shared_ptr<Node> compute_sequence_lengths(const Output<Node>& input_shape, int64_t batch_axis, int64_t seq_axis) {
+    auto batch_axis_const = make_shared<Constant>(element::i64, Shape{1}, batch_axis);
+    auto seq_axis_const = make_shared<Constant>(element::i64, Shape{1}, seq_axis);
+    auto gather_axis = make_shared<Constant>(element::i32, Shape{}, 0);
+    auto batch_dim = make_shared<Gather>(input_shape, batch_axis_const, gather_axis);
+    auto seq_dim = make_shared<Gather>(input_shape, seq_axis_const, gather_axis);
+    auto seq_lengths = make_shared<Broadcast>(seq_dim, batch_dim);
+
+    return seq_lengths;
+}
+
+OutputVector translate_reverse_base_op(const NodeContext& node,
+                                       const Output<Node>& input,
+                                       const std::vector<int64_t>& axes) {
+    // static rank for the input is required since the input can be a scalar and
+    // it is not clear how to define batch_axis and seq_axis that will be general
+    // for cases of multi-rank tensor and a scalar
+    auto input_rank = input.get_partial_shape().rank();
+    TENSORFLOW_OP_VALIDATION(
+        node,
+        input_rank.is_static(),
+        "OpenVINO TensorFlow Frontend does not support Reverse or ReverseV2 with input of undefined rank.");
+    auto input_rank_value = input_rank.get_length();
+
+    auto reverse_v2_node_name = node.get_name();
+    if (axes.size() == 0 || input_rank_value == 0) {
+        // there is nothing to reverse
+        input.get_tensor().add_names({reverse_v2_node_name + ":0"});
+        return {input};
+    }
+
+    TENSORFLOW_OP_VALIDATION(
+        node,
+        axes.size() == 1,
+        "OpenVINO TensorFlow Frontend does not support Reverse or ReverseV2 with multiple axes for the reversing.");
+
+    int64_t seq_axis = axes[0];
+    int64_t batch_axis = 0;
+
+    // normalize seq_axis if it is negative
+    seq_axis = (seq_axis < 0) ? seq_axis + input_rank_value : seq_axis;
```

Comment:
according to the spec https://docs.openvino.ai/latest/openvino_docs_ops_movement_ReverseSequence_1.html
seq_axis and batch_axis are allowed to be negative
Range of values : an integer within the range [-rank(data), rank(data) - 1]

Moreover, we can support dynamic `input_rank` is the information above is correct

Reply:
Let me refactor this solution. I think scalar case is not supported and not considered by TensorFlow. We can assume to have 1d-nD, based on it I can implement dynamic rank support.

Reply:
Fixed.